AI autofix editor summary

Changes made:
- Updated `prompts/mode-validate-generate.txt:207-210,226-227` to broaden graceful-shutdown guidance and detection from a single literal (`npm error signal SIGTERM`) to wrapper-variant markers (including `npm ERR!` and `Command failed with signal "SIGTERM"`), and made the grep case-insensitive/flexible to avoid false negatives for yarn/pnpm/npm output variants.
- Updated `prompts/mode-validate-fix-harness.txt:74-75` to align fix-harness instructions with the same broader wrapper-SIGTERM marker wording, so both validate prompts are consistent and do not imply an overly narrow log phrase match.
- Missing-context note: `AGENTS.md` (uppercase) was not present in repo root; used `agents.md` (lowercase) as the available app-level instruction file.

Already satisfied (suggested but already present):
- `prompts/mode-validate-generate.txt:218` already used `--filter "label=com.docker.compose.service=app"` fallback container lookup from the previous run, so no further container-filter change was needed.
- `prompts/mode-validate-generate.txt:224-234` already handled non-numeric/failed inspect via `|| echo unknown`, and `unknown` already fails the accept-list safely without being treated as graceful shutdown.

Ignored suggestions (with short reason):
- Skipped adding compose-project scoping to fallback `docker ps` filter because no stable project-name variable is defined in this prompt snippet; adding speculative env assumptions would broaden scope beyond minimal validated fix.
- Skipped changing `validation/logs/compose.log` contract to dynamic log discovery because this prompt already requires capturing logs under `validation/logs/`, and this run’s concrete correctness bug was regex narrowness (fixed).
- Skipped larger hardening proposal exceeding minimal-change intent (reviewer suggestion estimated ~15 lines); applied only small, direct, evidence-backed fixes.

Reviewer files processed:
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_deepseek_deepseek-v3_2.txt` — checksum `f22d99c2130566c76f506b1aab2344cd7eab7a5eb0fbd47cc168db425b14db61` — used (contained concrete line-referenced regex/container/log suggestions to validate).
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_google_gemini-3-flash-preview.txt` — checksum `2b59196183ddfe0e93de2e8ce72480971018b0ae35ae2fcf99e714e9469542e1` — used (contained concrete line-referenced findings; partially valid).
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_minimax_minimax-m2_5.txt` — checksum `90cbd41ea06881c77374a924416e6a15cb8eb991a0b6cba0e581cae50fcaad1c` — used (confirmation-only review; no actionable defect beyond already-satisfied context).
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_moonshotai_kimi-k2_5.txt` — checksum `77aa10cb6cee146366cdd1d544f6ea078cae195877abd67b2ee9b29f8f93e03a` — used (high-signal regex false-negative finding, validated and applied).
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_stepfun_step-3_5-flash.txt` — checksum `f53398270532fd033af0daa69358f556dd3a5b75d6d89226f564673b7e4baf3b` — used (independent corroboration of regex runtime-failure path).

Review file issue audit:
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_deepseek_deepseek-v3_2.txt` — total issues listed: 3 — issues applied: 1 — issues already applied: 1 — issues ignored: 1
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_google_gemini-3-flash-preview.txt` — total issues listed: 2 — issues applied: 0 — issues already applied: 1 — issues ignored: 1
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_minimax_minimax-m2_5.txt` — total issues listed: 0 — issues applied: 0 — issues already applied: 0 — issues ignored: 0
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_moonshotai_kimi-k2_5.txt` — total issues listed: 3 — issues applied: 1 — issues already applied: 1 — issues ignored: 1
- `/tmp/codex-pr-24344470375-1-21360/previous_reviews/review_stepfun_step-3_5-flash.txt` — total issues listed: 3 — issues applied: 1 — issues already applied: 1 — issues ignored: 1

PR comment audit:
- entry[0] `gemini-code-assist[bot]` — path: n/a, line: n/a — disposition: applied — reason: broad review comment requested more robust matching; implemented by broadening wording + regex.
- entry[1] `gemini-code-assist[bot]` — path: `prompts/mode-validate-fix-harness.txt`, line: 74 — disposition: applied — reason: updated fix-harness wording to include wrapper SIGTERM marker variants.
- entry[2] `gemini-code-assist[bot]` — path: `prompts/mode-validate-generate.txt`, line: 210 — disposition: applied — reason: updated descriptive text to explicitly include `npm ERR!` / `Command failed with signal "SIGTERM"` variants.
- entry[3] `gemini-code-assist[bot]` — path: `prompts/mode-validate-generate.txt`, line: 227 — disposition: applied — reason: replaced narrow regex with case-insensitive broader pattern covering wrapper variants.
- entry[4] `Copilot` — path: `prompts/mode-validate-generate.txt`, line: 228 — disposition: ignored — reason: compose-log existence concern is advisory here; prompt already includes log-capture requirement and file-existence guard.
- entry[5] `copilot-pull-request-reviewer[bot]` — path: n/a, line: n/a — disposition: applied — reason: concrete portion (broaden log matching robustness) was implemented.
- entry[6] `Copilot` — path: `prompts/mode-validate-generate.txt`, line: 211 — disposition: applied — reason: guidance/pattern mismatch resolved by making detection pattern match described equivalents.
- entry[7] `Copilot` — path: `prompts/mode-validate-fix-harness.txt`, line: 75 — disposition: ignored — reason: suggestion to redesign log-source reference would expand scope; no direct runtime defect proven in current prompt contract.

Files changed / commit status:
- prompts/mode-validate-fix-harness.txt
- prompts/mode-validate-generate.txt

### Serena MCP efficiency

| Metric | Value |
|--------|-------|
| Serena tool calls | 428 |
| File-based fallback ops | 739 |
| **Serena efficiency** | **37%** |
| Top Serena tools | `search_for_pattern` (70), `get_symbols_overview` (68), `find_symbol` (46), `find_referencing_symbols` (46), `replace_symbol_body` (46) |
| Est. tokens (with Serena) | ~395,200 |
| Est. tokens (without Serena) | ~606,300 |
| **Est. token savings** | **~211,100 (35%)** |

> Serena adoption is below the configured warning threshold (37% < 50%).

--- a/prompts/mode-validate-fix-harness.txt
+++ b/prompts/mode-validate-fix-harness.txt
@@ -71,8 +71,9 @@ CRITICAL — graceful shutdown container resolution:
 - A harness-initiated `docker compose stop` sends SIGTERM to PID 1. The test MUST treat ALL of the following as a graceful shutdown (pass):
     1. Exit code 0 (clean return from a SIGTERM handler).
     2. Exit code 143 = 128 + SIGTERM (signal-exit convention).
-    3. Exit code 1 when PID 1 is an `npm` / `yarn` / `pnpm` wrapper AND `validation/logs/compose.log` contains a wrapper SIGTERM marker (for example `npm ERR! signal SIGTERM` or `Command failed with signal "SIGTERM"`) for the wrapped command. npm translates signal termination of its child into exit 1 rather than propagating 143; this is a wrapper artifact, not an application crash.
+    3. Exit code 1 when PID 1 is an `npm` / `yarn` / `pnpm` wrapper AND `validation/logs/compose.log` (captured via `docker compose ... logs`) contains a wrapper SIGTERM marker (for example `npm ERR! signal SIGTERM` or `Command failed with signal "SIGTERM"`) for the wrapped command. npm translates signal termination of its child into exit 1 rather than propagating 143; this is a wrapper artifact, not an application crash.
 - If a graceful shutdown test fails with `app exited with code 1 (expected 0)` AND the compose log shows a wrapper SIGTERM marker for the wrapped command (e.g. `sh -c hardhat node ...`), the bug is in the assertion, not the app. Fix by broadening the accept-list above rather than by changing the app or the container entrypoint.
+- If `validation/logs/compose.log` is missing at assertion time, fix the harness to write `docker compose ... logs` output to that path before running the wrapper-SIGTERM grep.
 - If a graceful shutdown test fails with "app container id unavailable after stop", this is almost certainly caused by using `docker compose ps -q <service>` (without `--status exited` or `--all`) after stop. Fix by switching to the `--status exited` or `--all` variant.
 
 Output contract:

--- a/prompts/mode-validate-generate.txt
+++ b/prompts/mode-validate-generate.txt
@@ -224,7 +224,7 @@ CRITICAL — graceful shutdown container resolution:
     EXIT_CODE=$(docker inspect --format='{{.State.ExitCode}}' "$CID" 2>/dev/null || echo unknown)
     NPM_SIGTERM=0
     if [ "$EXIT_CODE" = "1" ] && [ -f "$COMPOSE_LOG" ] \
-       && grep -qiE '(npm( ERR!)?.*signal.*SIGTERM|Command failed with signal "SIGTERM")' "$COMPOSE_LOG"; then
+       && grep -qiE '((npm|yarn|pnpm).*signal.*SIGTERM|Command failed with signal.*SIGTERM)' "$COMPOSE_LOG"; then
       NPM_SIGTERM=1
     fi
     if [ "$EXIT_CODE" = "0" ] || [ "$EXIT_CODE" = "143" ] || [ "$NPM_SIGTERM" = "1" ]; then

--- a/prompts/mode-validate-generate.txt
+++ b/prompts/mode-validate-generate.txt
@@ -64,7 +64,7 @@ Docker Compose healthcheck YAML rules (CRITICAL):
 - Build/start via Docker Compose using the generated test file (`-f validation/docker-compose.test.yml`)
 - Health polling with explicit timeout
 - Execute generated scripts in `validation/tests/`
-- Capture logs under `validation/logs/`
+- Capture logs under `validation/logs/`, including a deterministic Compose log at `validation/logs/compose.log` (used by graceful-shutdown SIGTERM detection)
 - Always teardown via trap on EXIT using `docker compose -f validation/docker-compose.test.yml down -v --remove-orphans`
 - Exit 0 on pass and non-zero on failure
 - Emit a final machine-parseable JSON result object as the last line on stdout
@@ -222,6 +222,7 @@ CRITICAL — graceful shutdown container resolution:
       exit 1
     fi
     EXIT_CODE=$(docker inspect --format='{{.State.ExitCode}}' "$CID" 2>/dev/null || echo unknown)
+    docker compose -f "$COMPOSE_FILE" logs --no-color app > "$COMPOSE_LOG" 2>&1 || true
     NPM_SIGTERM=0
     if [ "$EXIT_CODE" = "1" ] && [ -f "$COMPOSE_LOG" ] \
        && grep -qiE '((npm|yarn|pnpm).*signal.*SIGTERM|Command failed with signal.*SIGTERM)' "$COMPOSE_LOG"; then

